fastestimator/network.py

```diff
@@ -712,10 +713,64 @@ def transform(self, data: Dict[str, Any], mode: str, epoch: int = 1) -> Dict[str
             (batch_data, prediction_data)
         """
         # Distribute multi-gpu data for processing
+        sub_sample = False
         strategy = tf.distribute.get_strategy()
         if isinstance(strategy, tf.distribute.MirroredStrategy):
+            batch_size, num_devices = get_batch_size(data), strategy.num_replicas_in_sync
+            if batch_size < num_devices:
+                data = self._fill_batch(data, num_devices - batch_size)
+                sub_sample = True
             data = next(iter(strategy.experimental_distribute_dataset(tf.data.Dataset.from_tensors(data))))
-        return super().transform(data, mode, epoch)
+        results = super().transform(data, mode, epoch)
+        if sub_sample:
+            results = self._subsample_data(results, batch_size)
+        return results
+
+    def _fill_batch(self, data: T, n: int) -> T:
+        """Fill data on batch dimension repeating the first n indices at the end.
+        Args:
+            data: The data to be filled.
+            The number of times to be repeated.
+        Returns:
+            Filled data.
+        """
+        if isinstance(data, dict):
+            return {key: self._fill_batch(val, n) for (key, val) in data.items()}
+        elif isinstance(data, list):
+            return [self._fill_batch(val, n) for val in data]
+        elif isinstance(data, tuple):
+            return tuple([self._fill_batch(val, n) for val in data])
+        elif isinstance(data, set):
+            return set([self._fill_batch(val, n) for val in data])
+        elif hasattr(data, "shape"):
+            paddings = [[0, n]] + [[0, 0] for _ in range(len(data.shape) - 1)]
+            if tf.is_tensor(data):
+                return tf.pad(data, paddings=paddings, mode="symmetric")
+            else:
+                return np.pad(data, pad_width=paddings, mode="symmetric")
```

Comment:
is it possible user would get torch tensors from dataset and stick them in here? or do we downcast to numpy always?

Reply:
np.pad works on all of them.  I have removed the tf if branch

Reply:
cause it gets turning onto a tf tensor again by the dataset?

Reply:
right.